File: edit.js
// Store uploaded image URLs
let uploadedImages = [];

// Handle multiple image uploads and display them in a collage (max 4 images)
document.getElementById('uploadBtn').addEventListener('click', function () {
    const fileInput = document.getElementById('parkingImages');
    const uploadedImagesCollage = document.getElementById('uploadedImagesContainer');

    // Check if files are selected
    if (fileInput.files && fileInput.files.length > 0) {
        const filesToShow = Array.from(fileInput.files).slice(0, 4); // Limit to 4 images

        filesToShow.forEach(file => {
            const storageRef = firebase.storage().ref(`parking_images/${file.name}`);

            storageRef.put(file).then(snapshot => {
                snapshot.ref.getDownloadURL().then(url => {
                    // Add image URL to the uploadedImages array
                    uploadedImages.push(url);

                    // Create the image preview in the collage
                    const imgContainer = document.createElement('div');
                    imgContainer.classList.add('image-container');

                    const img = document.createElement('img');
                    img.src = url;

                    const removeBtn = document.createElement('button');
                    removeBtn.textContent = 'Remove';
                    removeBtn.classList.add('remove-btn');

                    // Add event listener for removing image
                    removeBtn.addEventListener('click', function () {
                        uploadedImagesCollage.removeChild(imgContainer);
                        uploadedImages = uploadedImages.filter(imageUrl => imageUrl !== url);
                    });

                    imgContainer.appendChild(img);
                    imgContainer.appendChild(removeBtn);
                    uploadedImagesCollage.appendChild(imgContainer);
                }).catch(error => {
                    console.error("Error getting download URL:", error);
                });
            }).catch(error => {
                console.error("Error uploading image:", error);
            });
        });

        if (fileInput.files.length > 4) {
            alert("Only the first 4 images will be displayed in the collage.");
        }
    } else {
        alert("Please select images to upload.");
    }
});

// Handle form submission
document.getElementById("editParkingForm").addEventListener("submit", function (event) {
    event.preventDefault();  // Prevent the default form submission

    // Gather form data
    const formData = {
        name: document.getElementById("name").value,
        price: document.getElementById("price").value,
        address: document.getElementById("address").value,
        zone: document.getElementById("zone").value,
        city: document.getElementById("city").value,
        start_time: document.getElementById("startTime").value,
        end_time: document.getElementById("endTime").value,
        capacity_car: document.getElementById("capacityCar").value,
        capacity_adapted: document.getElementById("capacityAdapted").value,
        images: uploadedImages  // Add uploaded images URLs here
    };

    // Send the data to Firestore
    firebase.firestore().collection("parking_lots").add(formData)
    .then(() => {
        alert("Parking lot successfully added!");
    })
    .catch(error => {
        console.error("Error submitting form:", error);
        alert("Error submitting form");
    });
});

File: edit.py
import firebase_admin
from firebase_admin import credentials, firestore, storage
from flask import Flask, request, jsonify
from flask_cors import CORS
import os

# Initialize Firebase Admin SDK
try:
    cred = credentials.Certificate("service.json")  # Replace with the path to your Firebase service account key
    firebase_admin.initialize_app(cred, {
        'storageBucket': 'gs://webparking-c5e59.appspot.com'  # Replace with your Firebase Storage bucket URL
    })
    print("Firebase Admin SDK initialized successfully.")
except Exception as e:
    print(f"Error initializing Firebase Admin SDK: {e}")

# Initialize Firestore
firestore_db = firestore.client()
bucket = storage.bucket()  # Initialize Firebase Storage

# Initialize Flask app
app = Flask(__name__)
CORS(app)  # Enable CORS for all routes

def add_parking_lot_firestore(data):
    """
    Add parking lot data to Firestore.
    :param data: A dictionary containing the parking lot data.
    """
    try:
        firestore_db.collection('parking_lots').add(data)
        return "Parking lot data added successfully to Firestore."
    except Exception as e:
        return f"An error occurred in Firestore: {e}"

@app.route('/upload_image', methods=['POST'])
def upload_image():
    """
    Endpoint to upload images to Firebase Storage.
    """
    try:
        file = request.files['file']
        blob = bucket.blob(f"parking_images/{file.filename}")
        blob.upload_from_file(file, content_type=file.content_type)
        blob.make_public()  # Make the file public
        return jsonify({"image_url": blob.public_url}), 200
    except Exception as e:
        return jsonify({"error": str(e)}), 400

@app.route('/submit_parking_lot', methods=['POST'])
def submit_parking_lot():
    """
    Endpoint to receive parking lot data and upload it to Firebase Firestore.
    """
    try:
        # Parse JSON data from the request
        data = request.get_json()

        # Ensure required fields are provided
        required_fields = ['name', 'price', 'address', 'zone', 'city', 'start_time', 'end_time', 'capacity_car', 'capacity_adapted', 'images']
        for field in required_fields:
            if field not in data:
                return jsonify({"error": f"Missing required field: {field}"}), 400

        # Prepare the data for Firebase Firestore
        parking_lot_data = {
            'name': data.get('name'),
            'price': float(data.get('price')),
            'address': data.get('address'),
            'zone': data.get('zone'),
            'city': data.get('city'),
            'working_hours': {
                'start': data.get('start_time'),
                'end': data.get('end_time')
            },
            'capacity': {
                'car': int(data.get('capacity_car')),
                'adapted': int(data.get('capacity_adapted'))
            },
            'images': data.get('images')  # Handle uploaded image URLs
        }

        # Upload data to Firestore
        response = add_parking_lot_firestore(parking_lot_data)

        return jsonify({"message": response}), 200

    except Exception as e:
        return jsonify({"error": str(e)}), 400

if __name__ == '__main__':
    app.run(debug=True, host='0.0.0.0', port=5500)  # You can change the port as needed
